client/lease.go

```diff
@@ -0,0 +1,128 @@
+package client
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/uuid"
+	xlineapi "github.com/xline-kv/go-xline/api/xline"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+// Client for Lease operations
+type leaseClient struct {
+	// Name of the LeaseClient, which will be used in CURP propose id generation
+	name string
+	// The client running the CURP protocol, communicate with all servers.
+	curpClient curpClient
+	// The lease RPC client, only communicate with one server at a time
+	leaseClient xlineapi.LeaseClient
+	// Auth token
+	token string
+	// Lease Id generator
+	idGen leaseIdGenerator
+}
+
+// Creates a new `LeaseClient`
+func newLeaseClient(
+	name string,
+	curpClient curpClient,
+	innerLeaseClient xlineapi.LeaseClient,
+	token string,
+	idGen leaseIdGenerator,
+) leaseClient {
+	return leaseClient{
+		name:        name,
+		curpClient:  curpClient,
+		leaseClient: innerLeaseClient,
+		token:       token,
+		idGen:       idGen,
+	}
+}
+
+// Build inner client
+func buildLeaseClientFromAddrs(addrs []string, timeout ClientTimeout) (xlineapi.LeaseClient, error) {
+	conn, err := grpc.Dial(addrs[0], grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithIdleTimeout(timeout.idleTimeout))
```

Comment:
You can add load balancing to `addrs`. These changes can be updated in subsequent PRs.

Reply:
Okay, I'll update this in subsequent PRs.